CONTRIBUTING.md

```diff
@@ -70,20 +71,73 @@ When you are contributing to changelog, please follow these suggestions:
 
 3. Once you have completed your work, create a Pull Request, ensuring that it meets the requirements listed below.
 
-### Testing
+### Requirements for Pull Requests
+
+* Please create Pull Requests against the `master` branch.
+* Please make sure that your code complies with [PEP8](https://www.python.org/dev/peps/pep-0008/).
+* One line should not contain more than 100 characters.
+* Make sure that new code is covered by a test case (new or existing one).
+* We don't like [spaghetti code](https://en.wikipedia.org/wiki/Spaghetti_code).
+* All the tests in the test-suite have to pass.
+
+## Development Environment
+
+### Requirements
+
+Currently, the development environment can be set up on __Linux__, and __MacOS__.
+Before beginning the setup, please ensure you have installed and set up the following on your machine:
+
+- Docker ([installation guide](https://docs.docker.com/install/))
+- make utility
+
+### Building a test image
+
+Open up a terminal and go to the cloned `release-bot` git repository.
+Execute the command
+
+```
+make image-test
+```
+
+It builds a Docker image using your local repository, with release-bot installed in it. This may take some time when building for the first time.
+
+To check if the build was successful, run  `docker images`. If the build was successful, an image named `release-bot-tests` will be present in the output list.
+
+### Testing new changes
+
+To test any new changes you made in the code, first build a test image using the above instructions, and then execute the following command
+
+```
+docker run -it -v path/to/conf.yaml:/home/test-user/conf.yaml release-bot-tests /bin/bash
+```
+
+* `path/to/conf.yaml` is the absolute path of the configuration file in your machine.
+
+A terminal in the container will open up. Now run the bot using the command 
+
+```
+release-bot -d -c  conf.yaml
+```
+
+You will have to build the test image every time you make new changes that need to be tested.
+
+### Running the test-suite
 
 For testing, we are using [pytest](https://docs.pytest.org/en/latest/) framework. Tests are stored in the [tests](/tests) directory. We recommend to run tests inside docker container using:
+
 ```
 # execute whole test suite
-make test-in-container
+make test-in-container GITHUB_USER=your_github_username GITHUB_TOKEN=your_github_token
```

Comment:
GITHUB_USER is no longer needed, it's discovered from the token, we should also fix that in the makefile and jenkinsfile

Reply:
Created #191 for this